src/public/swagger-doc.json

```diff
@@ -177,7 +171,7 @@
         },
         "customerSig": {
           "type": "string",
-          "description": "The customer must sign the appointment to prove its authenticity. The appointment is encoded as a struct/tuple and combined with the pisa contract address before being hashed and signed.\n\n#### Example signature\n\n```\nconst appointmentBytes = ethers.utils.defaultAbiCoder.encode(\n  [\n    \"tuple(address,address,uint,uint,uint,bytes32,uint,bytes\"\n    + \"uint,uint,uint,address,string,bytes,bytes,bytes,bytes32)\",\n    \"address\"\"\n  ],\n  [\n    [\n      request.contractAddress,\n      request.customerAddress,\n      request.startBlock,\n      request.endBlock,\n      request.challengePeriod,\n      request.customerChosenId,\n      request.nonce,\n      request.data,\n      request.refund,\n      request.gasLimit,\n      request.mode,\n      request.eventAddress,\n      request.eventABI,\n      request.eventArgs,\n      request.preCondition,\n      request.postCondition,\n      request.paymentHash\n    ]\n    \"0xA02C7260c0020343040A504Ef24252c120be60b9\"\n  ]\n);\nconst hashForSig = ethers.utils.keccak256(appointmentBytes);\nconst customerSig = await wallet.signMessage(ethers.utils.arrayify(hashForSig));\n```\n",
+          "description": "The customer must sign the appointment to prove its authenticity. The appointment is broken down into three subsections to allow for easier decoding in solidity. These must be encoded separately before being hashed and signed.\n\n#### Example signature\n\n```\nconst basicBytes = ethers.utils.defaultAbiCoder.encode(\n  [\n    \"uint\", \n    \"uint\", \n    \"uint\", \n    \"uint\", \n    \"uint\", \n    \"uint\", \n    \"bytes32\"\n  ],\n  [\n    request.id,\n    request.nonce,\n    request.startBlock,\n    request.endBlock,\n    request.challengePeriod,\n    request.refund,\n    request.paymentHash\n  ]\n);\n\nconst callBytes = ethers.utils.defaultAbiCoder.encode(\n  [\n    \"address\", \n    \"address\", \n    \"uint\", \n    \"bytes\"\n  ],\n  [\n    request.contractAddress, \n    request.customerAddress, \n    request.gasLimit, \n    request.data\n  ]\n);\n\nconst conditionBytes = ethers.utils.defaultAbiCoder.encode(\n  [\n    \"address\", \n    \"bytes\", \n    \"bytes\", \n    \"bytes\", \n    \"uint\"\n  ],\n  [\n    request.eventAddress,\n    request.topics,\n    request.preCondition,\n    request.postCondition,\n    request.mode\n  ]\n);\n\nconst appointmentBytes = ethers.utils.defaultAbiCoder.encode(\n  [\n    \"bytes\", \n    \"bytes\", \n    \"bytes\"\n  ], \n  [\n    basicBytes, \n    callBytes,\n    conditionBytes\n  ]\n);\n\nconst hashForSig = ethers.utils.keccak256(appointmentBytes);\nconst customerSig = await wallet.signMessage(ethers.utils.arrayify(hashForSig));\n```\n",
```

Comment:
This documentation doesnt look right. We encode as a tuple now.

Reply:
Actually it wasn't just the documentation, as the encoding in alpha.test.ts wasn't correctly updated (in particular, I was passing the wrong data and format for the new `topics` argument).
Fixed in d969f2f (although I couldn't test it locally, as it runs on the live Pisa). Now checking how to fix the documentation accordingly.
As far as I can tell, this code does not have a counterparty

Reply:
No, it doesn't.

To be honest, just having to reformat that code occasionally is so painful that I'm considering changing our documentation.

Reply:
Actually, there seems to be a mismatch (even in master) between the encoding used in the client and the encoding used in the alpha test, see https://github.com/PISAresearch/pisa/blob/master/client/index.ts#L50 and https://github.com/PISAresearch/pisa/blob/master/test/smoke/alpha.test.ts#L10.